docs/fundamentals/executor/executor-in-flow.md

```diff
@@ -0,0 +1,462 @@
+	🔗 Protocol: 		GRPC
+	🏠 Local access:	0.0.0.0:54317
+	🔒 Private network:	192.168.1.187:54317
+	🌐 Public address:	212.231.186.65:54317
+executor0/rep-0@28271[E]:NotImplementedError('no time for it')
+ add "--quiet-error" to suppress the exception details
+Traceback (most recent call last):
+  File "/home/joan/jina/jina/jina/serve/runtimes/worker/__init__.py", line 101, in process_data
+    return await self._data_request_handler.handle(requests=requests)
+  File "/home/joan/jina/jina/jina/serve/runtimes/request_handlers/data_request_handler.py", line 94, in handle
+    field='docs',
+  File "/home/joan/jina/jina/jina/serve/executors/__init__.py", line 224, in __acall__
+    return await self.__acall_endpoint__(__default_endpoint__, **kwargs)
+  File "/home/joan/jina/jina/jina/serve/executors/__init__.py", line 231, in __acall_endpoint__
+    return await run_in_threadpool(func, self._thread_pool, self, **kwargs)
+  File "/home/joan/jina/jina/jina/helper.py", line 1200, in run_in_threadpool
+    executor, functools.partial(func, *args, **kwargs)
+  File "/usr/lib/python3.7/concurrent/futures/thread.py", line 57, in run
+    result = self.fn(*self.args, **self.kwargs)
+  File "/home/joan/jina/jina/jina/serve/executors/decorators.py", line 115, in arg_wrapper
+    return fn(*args, **kwargs)
+  File "/home/joan/jina/jina/toy.py", line 8, in foo
+    raise NotImplementedError('no time for it')
+NotImplementedError: no time for it
+NotImplementedError('no time for it')
+```
+
+## Gracefully close Executor
+
+You might need to execute some logic when your executor's destructor is called. For example, you want to
+persist data to the disk (e.g. in-memory indexed data, fine-tuned model,...). To do so, you can overwrite the
+method `close` and add your logic.
+
+Jina will make sure that the `close` method is executed when the `Executor` is terminated inside a Flow or when deployed in any cloud-native environment.
+
+One can think as if `Jina` would be using the `Executor` as a context manager making sure that the `close` method is always executed.
+
+```python
+from docarray import Document, DocumentArray
+from jina import Flow, Executor, requests
+
+
+class MyExec(Executor):
+
+    @requests
+    def foo(self, docs, **kwargs):
+        for doc in docs:
+            print(doc.text)
+
+    def close(self):
+        print("closing...")
+
+
+with MyExec() as executor:
+    executor.foo(DocumentArray([Document(text='hello world')]))
+
+f = Flow().add(uses=MyExec)
+
+with f:
+    f.post(inputs=DocumentArray([Document(text='hello world')])
+```
+
+```console
+Using MyExec as a context manager
+hello world
+closing...
+Using MyExec in the Flow
+           Flow@30104[I]:🎉 Flow is ready to use!
+	🔗 Protocol: 		GRPC
+	🏠 Local access:	0.0.0.0:54787
+	🔒 Private network:	192.168.1.203:54787
+	🌐 Public address:	212.231.186.65:54787
+hello world
+closing...
+
+Process finished with exit code 0
+```
+
+## Multiple DocumentArray as input
```

Comment:
```suggestion
## Multiple DocumentArrays as input
```

This section seems a bit disconnected from its title. When do I want to have multiple documentarrays as inputs? do i ever need to worry about that, or implement that myself? or can i just use a pre-made merger?

Reply:
It is just for completion. You can add info yourself maybe about pre defined merger